### PR TITLE
Update XSLDocument.java

### PR DESCRIPTION
**Pull request by imrandurranee**

### CODE REVIEW

**Comment by dmarkov** [comment]
@imrandurranee Let me find a reviewer for this pull request, thanks for submitting it

**Comment by dmarkov** [comment]
@pinaf please review, thanks

**Comment by pinaf** [comment]
@imrandurranee what issue is this PR solving? please provide a link to the issue in the first comment. thanks!

**Comment by imrandurranee** [comment]
This is to provide a custom error listener.

**Comment by pinaf** [comment]
@imrandurranee I"ll take that as there not being an original issue then. Please take a look at this travis error:
```
[WARNING] FindBugs: L P SIC: The class com.jcabi.xml.XSLDocument$1 could be refactored into a named _static_ inner class  At XSLDocument.java:[lines 214-226]
```

File: src/main/java/com/jcabi/xml/XSLDocument.java [hunk 1]
```diff
@@ -107,20 +107,7 @@
...
-        @Override
-        public void error(final TransformerException exception) {
-            Logger.error(this, exception.getMessageAndLocation());
-        }
-        @Override
-        public void fatalError(final TransformerException exception) {
-            this.error(exception);
-        }
-    };
+    private final transient ErrorListener elistener;
```
Comment (pinaf) [inline src/main/java/com/jcabi/xml/XSLDocument.java]: @imrandurranee let's rename this to `listener`

File: src/main/java/com/jcabi/xml/XSLDocument.java [hunk 2]
```diff
@@ -177,6 +164,21 @@ public XSLDocument(@NotNull(message = "XSL input stream can't be NULL")
...
 
     /**
+     * Public ctor, from XSL as a string with error listener.
+     * @param src XML document body
+     * @param elisteners Error Listener
+     */
+    public XSLDocument(@NotNull(message = "XSL can't be NULL")
+        final String src,
+        @NotNull(message = "XSL can't be NULL")
+        final ErrorListener elisteners) {
```
Comment (pinaf) [inline src/main/java/com/jcabi/xml/XSLDocument.java]: @imrandurranee rename to `lstnr`

File: src/main/java/com/jcabi/xml/XSLDocument.java [hunk 3]
```diff
@@ -209,6 +211,20 @@ public XSLDocument(
...
+            }
+            @Override
+            public void error(final TransformerException exception) {
+                Logger.error(this, exception.getMessageAndLocation());
+            }
+            @Override
+            public void fatalError(final TransformerException exception) {
+                this.error(exception);
+            }
+            };
```
Comment (pinaf) [inline src/main/java/com/jcabi/xml/XSLDocument.java]: @imrandurranee bad indentation here

File: src/main/java/com/jcabi/xml/XSLDocument.java [hunk 4]
```diff
@@ -177,6 +164,21 @@ public XSLDocument(@NotNull(message = "XSL input stream can't be NULL")
     }
 
     /**
+     * Public ctor, from XSL as a string with error listener.
+     * @param src XML document body
+     * @param elisteners Error Listener
+     */
+    public XSLDocument(@NotNull(message = "XSL can't be NULL")
```
Comment (pinaf) [inline src/main/java/com/jcabi/xml/XSLDocument.java]: @imrandurranee don't see the need for this constructor. no code is using it.

**Comment by pinaf** [comment]
@imrandurranee 3 comments. please make sure to address all your comments to someone (usually to me).

**Comment by pinaf** [comment]
@imrandurranee ping

**Comment by pinaf** [comment]
@imrandurranee hello? are you still working on this? If not, please close the issue.

**Comment by pinaf** [comment]
@yegor256 looks like this one was abandoned. can you please close it?

**Pull request closed without merging**